models/dataset_test.go

```diff
@@ -135,6 +142,62 @@ func TestCreateVersion(t *testing.T) {
 	})
 }
 
+func TestValidateDataset(t *testing.T) {
+	t.Parallel()
+
+	Convey("Unsuccessful validation (false) returned", t, func() {
+
+		Convey("when dataset.URI is unable to be parsed into url format", func() {
+			dataset := createDataset()
+			dataset.URI = ":foo"
+			fmt.Println(dataset.URI)
+			validationErr := ValidateDataset(testContext, &dataset)
+			So(validationErr, ShouldNotBeNil)
+			So(validationErr.Error(), ShouldResemble, errors.New("invalid fields: [URI]").Error())
+		})
+
+		Convey("when dataset.URI does not contain 'datasets' path", func() {
+			dataset := createDataset()
+			dataset.URI = "/123"
+			valid := ValidateDataset(testContext, &dataset)
```

Comment:
I think the error should be renamed from `valid` to `validationErr` (similarly to what you did above) - the name is much clearer (we know it's an error at first glance)